src/trackExtensions.ts

```diff
@@ -1,68 +1,184 @@
+import {
+  deleteExtensionFromEvent,
+  isExtensionDeprecated,
+  isExtensionInitialised,
+  mapLogToContractEvent,
+  toNumber,
+  verbose,
+  writeExtensionFromEvent,
+  writeExtensionVersionFromEvent,
+} from './utils';
+import { SUPPORTED_EXTENSION_IDS } from './constants';
+import stats from '../run/stats.json';
+import { extensionSpecificEventsListener } from './eventListener';
 
 export default async (): Promise<void> => {
-  verbose('Fetching already installed extensions');
+  const latestBlock = stats.latestBlock ?? 0;
 
-  supportedExtensionIds.forEach(async (extensionId) => {
-    const extensionInstalledLogs = await getLogs(
-      networkClient,
-      networkClient.filters.ExtensionInstalled(extensionId),
-    );
-    const extensionUninstalledLogs = await getLogs(
-      networkClient,
-      networkClient.filters.ExtensionUninstalled(extensionId),
+  SUPPORTED_EXTENSION_IDS.forEach(async (extensionId) => {
+    verbose(
+      `Fetching current version of extension: ${extensionId} starting from block: ${latestBlock}`,
     );
+    await trackExtensionAddedToNetwork(extensionId, latestBlock);
+
+    verbose(`Fetching events for extension: ${extensionId}`);
+    await trackExtensionEvents(extensionId, latestBlock);
+  });
+};
+
+const trackExtensionAddedToNetwork = async (
+  extensionId: string,
+  latestBlock: number,
+): Promise<void> => {
+  const extensionHash = getExtensionHash(extensionId);
+  const extensionAddedToNetworkLogs = await getLogs(
+    networkClient,
+    networkClient.filters.ExtensionAddedToNetwork(extensionHash),
+    { fromBlock: latestBlock },
+  );
+
+  // Only interested in the most recent one which contains the newest version
+  const mostRecentLog =
+    extensionAddedToNetworkLogs[extensionAddedToNetworkLogs.length - 1];
+  if (!mostRecentLog) {
+    return;
+  }
+
+  const event = await mapLogToContractEvent(
+    mostRecentLog,
+    networkClient.interface,
+  );
+  if (!event) {
+    return;
+  }
+
+  writeExtensionVersionFromEvent(event);
+};
+
+const trackExtensionEvents = async (
+  extensionId: Extension,
+  latestBlock: number,
+): Promise<void> => {
+  const extensionHash = getExtensionHash(extensionId);
+  const extensionInstalledLogs = await getLogs(
+    networkClient,
+    networkClient.filters.ExtensionInstalled(extensionHash),
+  );
+  const extensionUninstalledLogs = await getLogs(
+    networkClient,
+    networkClient.filters.ExtensionUninstalled(extensionHash),
+  );
+
+  /**
+   * Looping through the logs to create a mapping between colonies
+   * and the number of extension installations we encounter, minus uninstallations
+   */
+  const installedInColonyCount: { [address: string]: number } = {};
+  extensionInstalledLogs.forEach((log) => {
+    const parsedLog = networkClient.interface.parseLog(log);
+    const { colony } = parsedLog.args;
+
+    if (colony in installedInColonyCount) {
+      installedInColonyCount[colony]++;
```

Comment:
I'd would have wrote this as 
```suggestion
      installedInColonyCount[colony] += 1;
```

As it makes it more readable